lucene/core/src/java/org/apache/lucene/codecs/blocktree/CompressionAlgorithm.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.codecs.blocktree;
+
+import java.io.IOException;
+
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.util.compress.LowercaseAsciiCompression;
+
+/**
+ * Compression algorithm used for suffixes of a block of terms.
+ */
+enum CompressionAlgorithm {
+
+  NO_COMPRESSION(0x00) {
+
+    @Override
+    void read(DataInput in, byte[] out, int len) throws IOException {
+      in.readBytes(out, 0, len);
+    }
+
+    @Override
+    public String toString() {
+      return "no_compression";
+    }
+  },
+
+  LOWERCASE_ASCII(0x01) {
+
+    @Override
+    void read(DataInput in, byte[] out, int len) throws IOException {
+      LowercaseAsciiCompression.decompress(in, out, len);
+    }
+
+    @Override
+    public String toString() {
+      return "lowercase_ascii";
+    }
+  },
+
+  LZ4(0x02) {
+
+    @Override
+    void read(DataInput in, byte[] out, int len) throws IOException {
+      org.apache.lucene.util.compress.LZ4.decompress(in, len, out, 0);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
What's wrong with the default enum toString?

Reply:
I wanted to keep the string representations lowercased in Stats#toString like in the previous iteration of this pull request, but no strong feeling, I don't mind removing these `toString` implementations.

Reply:
I suggest then adding a default implementation that toLowerCase's name().  It would also be helpful to add a comment explaining where this is used; I didn't know.

Reply:
I switched back to the default implementation.